jcl/src/openj9.dtfjview/share/classes/com/ibm/jvm/dtfjview/commands/HexdumpCommand.java

```diff
@@ -37,161 +39,153 @@
+					lineBuffer.setLength(0);
 				}
-			}catch (MemoryAccessException e) {
-				out.println("Address not in memory - 0x" 
-						+ Long.toHexString(imagePointer.getAddress()));
-				return;
-			}catch (CorruptDataException e) {
+			} catch (CorruptDataException e) {
 				out.println("Dump data is corrupted");
 				return;
+			} catch (MemoryAccessException e) {
+				out.println("Address not in memory - 0x" + Long.toHexString(address + byteOffset));
+				return;
 			}
 		}
-		
-		long undisplayedBytes = 16 - i%16; 
-		if (16 != undisplayedBytes){
-			stringBuffer.append(padSpace(undisplayedBytes, asciiChars));
-			if (is_zOSdump){
-				// Add padding and output the remaining EBCDIC characters
-				for (int j = 0; j < undisplayedBytes; j++){
-					ebcdicChars = " " + ebcdicChars;
+
+		int lineOffset = numBytesToPrint % 16;
+
+		if (0 != lineOffset) {
+			for (int offset = lineOffset; offset < 16; ++offset) {
+				if (0 == (offset % 4)) {
+					lineBuffer.append(' ');
 				}
-				stringBuffer.append(" " + ebcdicChars);
+				lineBuffer.append("  ");
 			}
+
+			Arrays.fill(asciiBlock, lineOffset, 16, ' ');
+			lineBuffer.append("  |");
+			lineBuffer.append(asciiBlock);
+			lineBuffer.append('|');
+
+			if (is_zOSdump) {
+				Arrays.fill(ebcdicBlock, lineOffset, 16, ' ');
+				lineBuffer.append(" |");
+				lineBuffer.append(ebcdicBlock);
+				lineBuffer.append('|');
+			}
+
+			out.println(lineBuffer.toString());
 		}
-		stringBuffer.append("\n");
-		out.println(new String(stringBuffer));
-		
-		/*properties.put(Utils.CURRENT_MEM_ADDRESS, 
-				Long.toHexString(addressInDecimal+numBytesToPrint));*/
-		ctx.getProperties().put(Utils.CURRENT_MEM_ADDRESS, Long.valueOf(addressInDecimal));
+
+		out.println();
+
+		ctx.getProperties().put(Utils.CURRENT_MEM_ADDRESS, Long.valueOf(address));
 		ctx.getProperties().put(Utils.CURRENT_NUM_BYTES_TO_PRINT, Integer.valueOf(numBytesToPrint));
 	}
-	
-	private String padSpace(long undisplayedBytes, String asciiChars){
-		for (int i = 0; i < 2*undisplayedBytes + undisplayedBytes/4; i++){
-			asciiChars = " " + asciiChars;
-		}
-		return asciiChars;
-	}
-	
-	private String fixHexStringLength(String rawHexString){
-		int length = rawHexString.length();
-		if (1 == length){
-			return ("0" + rawHexString);
-		}
-		else if (2 == length){
-			return rawHexString;
-		}
-		else if (8 == length){
-			return rawHexString.substring(6, 8);
-		}
-		else{
-			return "ERROR fixHexStringLength";
-		}
-	}
-	
+
 	@Override
 	public void printDetailedHelp(PrintStream out) {
-		out.println("outputs a section of memory in hexadecimal, ascii and ebcdic\n\n" + 
-				"parameters: <hex_address> <bytes_to_print>\n\n" +
-				"outputs <bytes_to_print> bytes of memory contents " +
-				"starting from <hex_address>, ebcdic output is provided for z/OS dumps"
-				);
-		
+		out.format("outputs a section of memory in hexadecimal, ASCII and EBCDIC%n"
+				+ "%n" 
+				+ "parameters: <hex_address> <bytes_to_print>%n" 
```

Comment:
Seems `<bytes_to_print>` should have surrounding `[]` characters since it's optional. The following help could also mention the default of 256.

Reply:
Agreed.